network/p2p/libp2pNode.go

```diff
@@ -536,7 +536,7 @@ func (n *Node) UpdateAllowList(identities flow.IdentityList) error {
 	for i, identity := range identities {
 		allowlist[i], err = PeerAddressInfo(*identity)
 		if err != nil {
-			return fmt.Errorf("could not generate address info: %w", err)
+			n.logger.Err(err).Str("identity", identity.String()).Msg("could not generate address info")
```

Comment:
Not returning an error but rather logging it.

Reply:
I think this would cause `allowlist` to have `nil` object.

Reply:
yup, changed.

Reply:
Thanks.

Another comment is that I wonder if we could do this check earlier?

Since the validation of the peer address does not depend on any state, we could technically move the validation earlier. We just need to figure out the responsibility of who should validate the identity here. libp2p2? or overlay? 

libp2p receives this list from [`overlay.Identity`](https://github.com/onflow/flow-go/blob/a551c98add8a71d921b4339080cded64038e9efb/network/p2p/middleware.go#L427), should we let overlay filter out peers with invalid address?

Reply:
yeah, that is not a bad idea. @smnzhu is actually replacing the `ids` with an ID provider so I will add this validation over there.